std/experimental/allocator/building_blocks/bucketizer.d

```diff
@@ -82,7 +82,7 @@ struct Bucketizer(Allocator, size_t min, size_t max, size_t step)
         {
             const actual = goodAllocSize(bytes);
             auto result = a.alignedAllocate(actual, alignment);
-            return result !is null ? (() @trusted => (&result)[0 .. bytes])() : null;
+            return result !is null ? (() @trusted => (&result[0])[0 .. bytes])() : null;
```

Comment:
Took me a while to discover this bug, as, imo, it's easy to miss 😄

Reply:
Can you explain for my benefit?

Reply:
The intent here is to return a slice of the underlying memory chunk (`result.ptr`). In other words, we want `result.ptr[0 .. bytes]`.

The bug is that `&result` is different from `result.ptr`; `&result` takes the address of the struct that holds the actual raw pointer and the length information.

The equivalent of `result.ptr` is `&result[0]`.

Reply:
Okay, thought that was it but wasn't sure. I'm surprised that the struct's layout doesn't have the pointer first so this will Just Work.